configurator/sshkey/sshkey.go

```diff
@@ -0,0 +1,106 @@
+package sshkey
+
+import (
+	"encoding/json"
+	"io"
+	"io/ioutil"
+	"os"
+	"os/user"
+	"path"
+	"path/filepath"
+	"strconv"
+
+	"github.com/percona/pmm-manage/configurator/config"
+	log "github.com/sirupsen/logrus"
+	"golang.org/x/crypto/ssh"
+	"golang.org/x/sys/unix"
+)
+
+// PMMConfig pass configuration via global variable :'(
+var PMMConfig config.PMMConfig
+
+func RunSSHKeyChecks() {
+	sshKeyUser, err := user.Lookup(PMMConfig.SSHKeyOwner)
+	if err != nil {
+		log.Fatal(err)
+	}
+	if PMMConfig.SSHKeyPath == "" {
+		PMMConfig.SSHKeyPath = path.Join(sshKeyUser.HomeDir, ".ssh/authorized_keys")
+	}
+
+	sshKeyDir := filepath.Dir(PMMConfig.SSHKeyPath)
+	if dir, err := os.Stat(sshKeyDir); err != nil || !dir.IsDir() {
+		if err := os.MkdirAll(sshKeyDir, 0700); err != nil {
```

Comment:
Just clean on func name: https://golang.org/pkg/os/#MkdirAll. You will get `ENOTDIR`.